content/asp.net/getting-started/controllers-actions.md

```diff
@@ -1,20 +1,122 @@
+
+```c#
+    [Route("peeps")]
+    public class PersonController : Controller
+    {
+        [Route("favs")]
+        public IActionResult Favorites()
+        {
+            //...
+        }
+
+        [Route("/people/home")]
+        public IActionResult Index()
+        {
+            //...
+        }
+    }
+```
+
+Routes can get tricky to debug, so use them judiciously and know how they affect incoming requests in your app.  For example, someone looking at the above controller might make assumptions about the `Index` action, given the default route confiruration on the project; however, with the attributes applied as above, a request to `/person/index` would give them a `404 Not Found`.  Read more about routes [here](routing.md). 
+
+## Controller Helpers and Result Objects 
+In addition to the actions that you write yourself, you're also privy to the host of helper methods that are built into the base controller that you inherit from. These methods help to formulate a repsonse after you've completed any validation, service access or processing required. We've already looked at one as part of the samples above, but let's have another quick look:
+
+```c#
+public IActionResult Index()
+{
+    return View();
+}
+```        
+
+The call to the helper method `View` above gives you an easy way to say, "I'm done processing here, and now I would like the user to see the page related to this action."  The framework takes over, maintains the state of any properties you've set on the controller and renders a view following the default naming and location conventions we covered earlier.
+
+Just like `View`, there are other helpers that will make the lifting easier on the controller.
+
+ - **Redirection**- You can use `Redirect`, `LocalRedirect`, `RedirectToRoute` and other similar methods in your actions for the purpose of forwarding users on to other location, for example, after a login.   
+ - **Standard Response Types**- When you are working at more of an API level or need more granular control over building the response, you can use pre-built classes like `OkResult` or `NotFoundResult` to convey the results of your processing.   
+ - **Json**- If you want to control the way your data is returned, perhaps decorating an object in order to better interoperate with another service or client, you can manually build JSON responses. 
+ - **PartialView**- When you only need to render part of a page without the overhead of a layout, typically used in AJAX scenarios where only a portion of the page is fetched from the server.   
+ - **ViewBag, ViewData**- These are containers that are backed by the same key/value pairs in a dictionary. `ViewData` uses a string for a key, whereas `ViewBag` is a `dynamic` that allows you to the dictionary as though the keys were property names.  
```

Comment:
Not sure these fit, since they're not helpers but properties. But you can probably leave them.